File: 2015/Day_04/Python/test_part_1.py
from . import part_1


def test_calculate_hash():
    assert part_1.calculate_hash("abcdef609043") == "000001dbbfa3a5c83a2d506429c7b00e"


def test_does_it_lead():
    assert part_1.does_it_lead_with_five_zeroes("000001dbbfa3a5c83a2d506429c7b00e") is True
    assert part_1.does_it_lead_with_five_zeroes("00001dbbfa3a5c83a2d506429c7b00e") is False


def test_find_special_number():
    assert part_1.find_special_number("abcdef") == 609043
    assert part_1.find_special_number("pqrstuv") == 1048970

File: 2015/Day_04/Ruby/test_part_1.rb
require_relative "part_1"
require 'test/unit'


class TestAdventCoinHash < Test::Unit::TestCase
    
    def test_create_hash
        assert_equal('000001dbbfa3a5c83a2d506429c7b00e', create_hash('abcdef609043'))
    end
    
    def test_start_with_five_zeroes
        assert_equal(true, start_with_five_zeroes("000001dbbfa3a5c83a2d506429c7b00e"))
        assert_equal(false, start_with_five_zeroes("00001dbbfa3a5c83a2d506429c7b00e"))
    end
    
    def test_find_number
        assert_equal(609043, find_number("abcdef"))
    end
end
